src/main/scala/io/kaizensolutions/virgil/CQL.scala

```diff
@@ -0,0 +1,143 @@
+package io.kaizensolutions.virgil
+
+import io.kaizensolutions.virgil.Proofs._
+import io.kaizensolutions.virgil.codecs.Reader
+import io.kaizensolutions.virgil.configuration.ExecutionAttributes
+import io.kaizensolutions.virgil.dsl.{Assignment, Relation}
+import zio._
+
+final case class CQL[+Result] private (
+  private[virgil] val cqlType: CQLType[Result],
+  private[virgil] val executionAttributes: ExecutionAttributes
+) { self =>
+  def +(that: CQL[MutationResult])(implicit ev: Result <:< MutationResult): CQL[MutationResult] = {
+    val resultCqlType =
+      (self.widen[MutationResult].cqlType, that.cqlType) match {
+        case (a: CQLType.Mutation, b: CQLType.Mutation) =>
+          CQLType.Batch(NonEmptyChunk(a, b), BatchType.Logged)
+
+        case (a: CQLType.Mutation, CQLType.Batch(mutations, thatBatchType)) =>
+          CQLType.Batch(a +: mutations, thatBatchType)
+
+        case (CQLType.Batch(mutations, batchType), b: CQLType.Mutation) =>
+          CQLType.Batch(mutations :+ b, batchType)
+
+        case (CQLType.Batch(m1, b1), CQLType.Batch(m2, b2)) =>
+          CQLType.Batch(m1 ++ m2, b1.combine(b2))
+
+        case (b @ CQLType.Batch(_, _), _) =>
+          b
+
+        case (_, b @ CQLType.Batch(_, _)) =>
+          b
+
+        // You cannot actually construct a Query[MutationResult]
+        case (m: CQLType.Mutation, _) =>
+          m
+
+        case (_, m: CQLType.Mutation) =>
+          m
+
+        case (a, _) =>
+          a
+      }
+    CQL(resultCqlType, self.executionAttributes.combine(that.executionAttributes))
+  }
+
+  def batchType(in: BatchType)(implicit ev: Result <:< MutationResult): CQL[MutationResult] =
```

Comment:
Consider also supporting this syntax:

```scala
CQL.unbatched { ... }
```

Prefix notation can be easier to read with region-based operators.

Reply:
And since there are only 3 types, you can special case the 3 constructors.

Reply:
Hey John, I like the idea of `CQL.unbatched`:

```scala
val individualMutations: NonEmptyChunk[CQL[MutationResult]] = CQL.unbatched { ... }
```

I was wondering if it makes sense to use an HList and track each of the mutations inside instead of using a NonEmptyChunk so we can give the user a more meaningful API allowing them to access the individual mutations or would that be overkill? 

I was envisioning something like this:
```scala
// given batchMutation consists of 3 individual mutations
val m1 :*: m2 :*: m3 :*: End: NonEmptyChunk[CQL[MutationResult]] = CQL.unbatched { batchMutation }
```

Reply:
> And since there are only 3 types, you can special case the 3 constructors.

Sorry, can you elaborate on this? do you want to make CQL into a sum type and embed/eliminate CQLType? This would result in Execution Attributes being defined in each subtype which is not a big deal or did you mean something else

Reply:
John showed me how to do this, changes are in